server/src/main/java/org/opensearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -669,6 +669,11 @@ private void createShard(DiscoveryNodes nodes, RoutingTable routingTable, ShardR
         try {
             final long primaryTerm = state.metadata().index(shardRouting.index()).primaryTerm(shardRouting.id());
             logger.debug("{} creating shard with primary term [{}]", shardRouting.shardId(), primaryTerm);
+            DiscoveryNode targetNode = nodes.getLocalNode();
+            // Set remote attributes on Shard routing if the target node for the shards has remote attributes
+            if (targetNode.isRemoteStoreNode()) {
+                shardRouting.setAssignedToRemoteStoreNode(true);
```

Comment:
appears to be error prone, is there a way to have this as final and set inside ctor?